geomesa-blobstore/geomesa-blobstore-api/src/main/scala/org/locationtech/geomesa/blob/api/BlobstoreServlet.scala

```diff
@@ -8,55 +8,107 @@
 
 package org.locationtech.geomesa.blob.api
 
-import com.typesafe.scalalogging.LazyLogging
-import org.locationtech.geomesa.accumulo.data.{AccumuloDataStore, AccumuloDataStoreFactory}
-import org.locationtech.geomesa.blob.core.AccumuloBlobStore
-import org.locationtech.geomesa.web.core.GeoMesaScalatraServlet
-import org.scalatra.{NotFound, Ok}
+import java.io.{File, IOException}
+import java.nio.file.attribute.PosixFilePermission._
+import java.nio.file.attribute.PosixFilePermissions
+import java.util.UUID
+
+import org.apache.commons.io.FilenameUtils
+import org.locationtech.geomesa.blob.core.AccumuloBlobStore._
+import org.locationtech.geomesa.utils.cache.FilePersistence
+import org.scalatra._
+import org.scalatra.servlet.{FileUploadSupport, MultipartConfig, SizeConstraintExceededException}
 
 import scala.collection.JavaConversions._
+import scala.collection.mutable
+import scala.util.Try
 
-class BlobstoreServlet extends GeoMesaScalatraServlet with LazyLogging {
+class BlobstoreServlet(val persistence: FilePersistence) extends GeoMesaPersistentBlobStoreServlet with FileUploadSupport with GZipSupport {
   override def root: String = "blob"
 
-  var abs: AccumuloBlobStore = null
-
-  // TODO: Revisit configuration and persistence of configuration.
-  // https://geomesa.atlassian.net/browse/GEOMESA-958
-  // https://geomesa.atlassian.net/browse/GEOMESA-984
-  post("/ds") {
-
-    logger.debug("In ds registration method")
-
-    val dsParams = datastoreParams
-    val ds = new AccumuloDataStoreFactory().createDataStore(dsParams).asInstanceOf[AccumuloDataStore]
+  val maxFileSize: Int = Try(System.getProperty(BlobstoreServlet.maxFileSizeSysProp, "50").toInt).getOrElse(50)
+  val maxRequestSize: Int = Try(System.getProperty(BlobstoreServlet.maxRequestSizeSysProp, "100").toInt).getOrElse(100)
+  // caps blob size
+  configureMultipartHandling(
+    MultipartConfig(
+      maxFileSize = Some(maxFileSize * 1024 * 1024),
+      maxRequestSize = Some(maxRequestSize * 1024 * 1024)
+    )
+  )
+  error {
+    case e: SizeConstraintExceededException =>
+      handleError("Uploaded file too large!", e)
+    case e: IOException =>
+      handleError("IO exception in BlobstoreServlet", e)
+  }
 
-    if (ds == null) {
-      NotFound(reason = "Could not load data store using the provided parameters.")
+  delete("/:id/?") {
+    val id = params("id")
+    logger.debug("In DELETE method for blobstore, attempting to delete: {}", id)
```

Comment:
picky can we just redo the comments to say "Attemping to delete: {}" and drop the "In DELETE method" part...same for other "In ID method" etc.